android/mobile/app/src/main/java/com/ttd/sso/sample/ui/AuthScreenViewModel.kt

```diff
@@ -0,0 +1,127 @@
+
+    private val _viewState = MutableStateFlow<AuthScreenState>(SignedOutState)
+    override val viewState: StateFlow<AuthScreenState> = _viewState.asStateFlow()
+
+    private var client: WebSignInFlowClient? = null
+    private val manager =  OpenPassManager.getInstance()
+
+    init {
+        // Observe the state of the OpenPassManager and translate those into our own ViewState.
+        viewModelScope.launch {
+            manager.state.collect { state ->
+                Log.d(TAG, "State Update: $state")
+
+                when (state) {
+                    is OpenPassManagerState.Loading -> Unit
+                    is OpenPassManagerState.SignedOut -> _viewState.emit(SignedOutState)
+                    is OpenPassManagerState.SignedIn -> state.tokens.idToken?.email?.let { _viewState.emit(SignedInState(it)) }
+                    is OpenPassManagerState.Error -> {
+                        _viewState.emit(ErrorState(state.error))
+                    }
+                }
+            }
+        }
+    }
+
+    override fun processAction(action: AuthScreenAction) {
+        Log.d(TAG, "Action: $action")
+
+        viewModelScope.launch {
+            when (action) {
+                is SignIn -> signIn(action.context)
+                is CheckSignIn -> checkSignIn(action.intent)
+                is SignOut -> signOut()
+            }
+        }
+    }
+
+    private suspend fun signIn(context: Context) {
+        client = WebSignInFlowClient(CLIENT_ID, manager)
+        client?.let { client ->
+            // ... Configure any theming
+
+            // Launch the sign in flow.
+            client.launchSignIn(context)
+
+            // Observe the results of the flow, handling them accordingly.
+            client.state.collect { state ->
+                when (state) {
+                    is WebSignInFlowState.Error -> _viewState.emit(ErrorState(state.error))
+                    is WebSignInFlowState.Complete -> {
+                        val tokens = manager.currentTokens
+                        if (tokens != null) {
+                            // Email Available
+                            tokens.idToken?.email?.let {
+                                _viewState.emit(SignedInState(it))
+                            }
+                        } else {
+                            // Tokens Not Available
+                            _viewState.emit(SignedOutState)
+                        }
+                    }
+                    is WebSignInFlowState.Launched -> Unit
+                }
+            }
+        }
+    }
+
+    private fun checkSignIn(intent: Intent) {
+        client?.checkSignIn(intent)
+    }
+
+    private fun signOut() {
+        client?.cancel()
+        client = null
+        manager.signOut()
+        _viewState.tryEmit(SignedOutState)
+    }
+
+    private companion object {
+        const val TAG = "AuthScreenViewModel"
+        const val CLIENT_ID = "d12d2cde12db4747ba7b555a8eb9defb"
```

Comment:
I am discussing with @dcaunt about this and the changes we should likely do to the SDK to avoid this. For now though, this is how it's done to keep things simple (avoiding adding gradle config with manifest metadata) to the sample. We will revisit